prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -2310,36 +2241,7 @@ subroutine initProgramVariables(input, env)
       EfieldPhase = 0
     end if
 
-    allocate(qInput(orb%mOrb, nAtom, nSpin))
-    allocate(qOutput(orb%mOrb, nAtom, nSpin))
-    qInput(:,:,:) = 0.0_dp
-    qOutput(:,:,:) = 0.0_dp
-
-    if (tMixBlockCharges) then
-      allocate(qBlockIn(orb%mOrb, orb%mOrb, nAtom, nSpin))
-      allocate(qBlockOut(orb%mOrb, orb%mOrb, nAtom, nSpin))
-      qBlockIn(:,:,:,:) = 0.0_dp
-      qBlockOut(:,:,:,:) = 0.0_dp
-      if (tImHam) then
-        allocate(qiBlockIn(orb%mOrb, orb%mOrb, nAtom, nSpin))
-        qiBlockIn(:,:,:,:) = 0.0_dp
-      end if
-    end if
-
-    if (tImHam) then
-      allocate(qiBlockOut(orb%mOrb, orb%mOrb, nAtom, nSpin))
-      qiBlockOut(:,:,:,:) = 0.0_dp
-    end if
-
-    if (tSccCalc) then
-      allocate(qDiffRed(nMixElements))
-      allocate(qInpRed(nMixElements))
-      allocate(qOutRed(nMixElements))
```

Comment:
As its removed here it now seems not be be allocated anywhere, so causes a segfault with gnu 7.4 later on in the reduce charges routine.
 ctest -R H3_skipcheck
is one of the tests then failing.

Reply:
Ah, should be before the if (.not. tReadChrg) then block.

Reply:
Resolved in commit [71c57690](https://github.com/AlexBuccheri/dftbplus/commit/71c57690b7033e08918833e1660fe70d714820b6)